src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -293,12 +293,12 @@ class ShenandoahHeap : public CollectedHeap, public ShenandoahSpaceInfo {
 
   size_t _gc_no_progress_count;
 
-  void set_gc_state_mask(uint mask, bool value);
+  void set_gc_state(uint mask, bool value);
 
 public:
   char gc_state() const;
-  void set_gc_state_all_threads();
-  bool has_gc_state_changed() { return _gc_state_changed; }
+  void propagate_gc_state_to_java_threads();
+  bool has_gc_state_changed() const { return _gc_state_changed; }
```

Comment:
The name changes per @shipilev look good. If we have a sentence of documentation in the header file for each of these methods, that makes things even clearer (and is not subject to limitations on the length of the method name :-)